providers/rfc2136.go

```diff
@@ -0,0 +1,182 @@
+package providers
+
+import (
+	"github.com/Sirupsen/logrus"
+	rdns "github.com/rancher/external-dns/dns"
+	"os"
+	"fmt"
+	"github.com/miekg/dns"
+	"strings"
+	"net"
+	"time"
+)
+
+type RFC2136Handler struct {
+	server     string
+	port       string
+	keyname    string
+	key        string
+	rootdomain string
+}
+
+/* Private functions */
+func needFQDN(domain string) string {
+	if (strings.HasSuffix(domain, ".")) {
+		return domain
+	} else {
+		return domain + "."
+	}
+}
+
+func (b RFC2136Handler) sendMessage(msg *dns.Msg) error {
+	t := new(dns.Transfer)
```

Comment:
Using the zone transfer method to send DNS Update messages is definitely not correct. It should be something along the lines of this:

```Go
func (b RFC2136Handler) sendMessage(msg *dns.Msg) error {
	c := new(dns.Client)
	c.TsigSecret = map[string]string{b.keyname: b.key}
	c.SingleInflight = true
	msg.SetTsig(b.keyname, dns.HmacMD5, 300, time.Now().Unix())
	nameserver := net.JoinHostPort(b.server, b.port)

	r, _, err := c.Exchange(msg, nameserver)
	if err != nil {
		return err
	}

	if r != nil && r.Rcode != dns.RcodeSuccess {
		return fmt.Errorf("Bad return code: %s", dns.RcodeToString[r.Rcode])
	}

	return nil
}
```